Source/JavaScriptCore/parser/Parser.cpp

```diff
@@ -772,6 +772,8 @@ template <class TreeBuilder> TreeStatement Parser<LexerType>::parseStatementList
         FALLTHROUGH;
     case AWAIT:
     case YIELD: {
+        semanticFailIfTrue(match(YIELD) && m_parserState.isParsingClassStaticBlock, "Cannot use 'yield' within class static block");
```

Comment:
Is it correct? What happens if we define a function inside static class block and we use yield inside it?

```
class A {
    static {
         function* gen() {
            yield 42;
         }
    }
}
```

Reply:
Yes, it should be correct. Thanks.

Reply:
And let's add tests for this.

Reply:
Yes, it's added.

Reply:
Let's remove `isParsingClassStaticBlock`. Having this state in a global ParserState is not a great way to handle the issue.